include/http/middleware/index.js

```diff
@@ -0,0 +1,346 @@
+        next();
+    };
+
+    Middleware.localizedRouteCheck = function (req, res, next) {
+        var pathVars = req.pathVars;
+        if (typeof pathVars.locale !== 'undefined') {
+            if (!this.siteObj.supportedLocales[pathVars.locale]) {
+
+                //TODO make this check more general
+                return req.handler.serve404();
+            }
+
+            //update the localization
+            req.handler.localizationService = req.localizationService = req.handler.deriveLocalization({
+                session: this.session,
+                routeLocalization: pathVars.locale
+            });
+        }
+        next();
+    };
+
+    Middleware.instantiateController = function (req, res, next) {
+
+        var rt = req.routeTheme;
+        var ControllerType = req.route.themes[rt.site][rt.theme][rt.method].controller;
+        req.controllerInstance = new ControllerType();
+
+        next();
+    };
+
+    Middleware.parseRequestBody = function (req, res, next) {
+        req.handler.parseBody(req.themeRoute.request_body, function (err, body) {
+            if (util.isError(err)) {
+                err.code = 400;
+            }
+            req.body = body;
+            next(err);
+        });
+    };
+
+    Middleware.initializeController = function (req, res, next) {
+        var props = {
+            request_handler: req.handler,
+            request: req,
+            response: res,
+            session: req.session,
+            localization_service: req.localizationService,
+            path_vars: req.pathVars,
+            pathVars: req.pathVars,
+            query: req.handler.url.query,
+            body: req.body,
+            site: req.site,
+            siteObj: req.siteObj,
+            siteName: req.siteName,
+            activeTheme: req.activeTheme,
+            routeLocalized: !!req.routeTheme.localization
+        };
+        req.controllerInstance.init(props, next);
+    };
+
+    Middleware.render = function (req, res, next) {
+        req.controllerInstance[req.themeRoute.handler ? req.themeRoute.handler : 'render'](function (result) {
+            if (util.isError(result)) {
+                return next(result);
+            }
+            req.controllerResult = result;
+            next();
+        });
+    };
+
+    Middleware.writeSessionCookie = function (req, res, next) {
+        var cookies = new Cookies(req, res);
+        if (req.setSessionCookie) {
+            try {
+                cookies.set(pb.SessionHandler.COOKIE_NAME, req.session.uid, pb.SessionHandler.getSessionCookie(req.session));
+            }
+            catch (e) {
+                pb.log.error('RequestHandler: Failed to set cookie: %s', e.stack);
+            }
+        }
+        next();
+    };
+
+    Middleware.writeResponse = function (req, res, next) {
+        var data = req.controllerResult;
+        var doRedirect = typeof data.redirect !== 'undefined';
```

Comment:
Is this variable necessary, as it's not reused and the check could just as well be done in the if-statement itself?